targets/targets.json

```diff
@@ -11,7 +11,17 @@
         "detect_code": [],
         "public": false,
         "default_lib": "std",
-        "bootloader_supported": false
+        "bootloader_supported": false,
+        "config": {
+            "console-uart-flow-control": {
+                "help": "Console uses hardware flow control. Default setting is disabled.",
+                "value": false
+            },
+            "console-uart-flow-control-type": {
+                "help": "Console hardware flow control type. Value must be of type SerialBase::Flow.",
+                "value": "SerialBase::RTSCTS"
```

Comment:
Yay, looks like the correct place.

I'd prefer to have a single setting here, rather than setting + enable. Not sure what @theotherjimmy thinks.

We should avoid the "SerialBase::" prefix - just enumerate the valid token-type strings, and the C++ code can add prefixes or otherwise detect as required.

If you ever need special compile-time behaviour for certain settings rather than simple passthrough, using simple identifiers makes that possible. (Compare config changes done in #6566 for channel band names)

So one option with valid values `null`, `"RTS"`, `"CTS"` and `"RTSCTS"`? Then for the current simple case you just  `#ifdef` on that, rather than `#if` on the enable, and it leaves open the door for future concat-type trickery. If you did that trickery now, you could specifically detect `"Disabled"`, but seems pointless.